README.md

```diff
@@ -5,7 +5,128 @@
+```
+"EventPropagation": {
+    "TopicName": "...",
+    "TopicAccessKey": "...",
+    "TopicEndpoint": "http://topicurl.domain.com/api/events"
+  }
+```
+
+To publish an event, use the IEventPropagationClient interface (via injection). Use the PublishDomainEventAsync to publish the event. The required Subject field is a string description to provide context for the event.
+
+```
+var coolCandorEvent = new ExampleDomainEvent
+{
+    CoolId = Guid.NewGuid().ToString(),
+    CoolDate = DateTime.UtcNow
+};
+
+await this._eventPropagationClient.PublishDomainEventAsync(subject: "TestEventPublication", coolCandorEvent);
+```
+
+#### Restrictions
+* Your service can publish events on only 1 topic.
+* The topic the service publishes to must be in the Topics class provided by the Workleap.EventPropagation package.
+
+#### Exceptions
+
+In case of error while publishing an event, an exception of type EventPropagationPublishingException is thrown. It provides the TopicName, Subject and TopicHostName fields.
+
+If configuration is bad, an exception of type InvalidConfigurationException is thrown. If the provided topic is bad, an exception of type InvalidTopicException is thrown.
+
+### Using the Workleap.EventPropagation package to subscribe to events
+
+When using dotnet core, you can register event propagation subscriptions at startup in the service collection.  To configure the subscriber, the list of subscribed topics is required.
+
+```
+services
+    .AddEventPropagationSubscriber()
+    .AddDomainEventHandlersFromAssembly(Assembly.GetExecutingAssembly())
+    .Configure(new[] { Topics.Organisation, Topics.Candor });
+```
+
+An API endpoint must be made available to receive events from the EventGrid topic. The package provides this endpoint for dotnet core projects. It can be added like so.
+```
+services.AddControllers().PartManager.ApplicationParts.Add(new AssemblyPart(typeof(EventGridController).Assembly));
+```
+
+#### Define domain event handlers
+
+You must define domain event handlers for events you wish to handle. You are not obligated to handle every possible event from a topic.
+
+Domain event handlers must implement the IDomainEventHandler<> interface.
+
+```
+public class ExampleDomainEventHandler : IDomainEventHandler<ExampleDomainEvent>
+{
+    public Task HandleDomainEventAsync(ExampleDomainEvent domainEvent)
+    {
+        return Task.CompletedTask;
+    }
+}
+```
+To register domain event handlers, you can either use the package’s extension to scan your assembly and register all found domainevent handlers, or register your favorite domain event handlers manually.
+```
+// dotnet core service collection extension: scan assembly for all domain event handlers
+.AddDomainEventHandlersFromAssembly(Assembly.GetExecutingAssembly())
+
+// dotnet core service collection extension: add single domain event handler
+.AddDomainEventHandler<ExampleDomainEventHandler>()
+```
```

Comment:
This section seems a bit redundant with the one above, we could probably combine them?